Add Pixi development environment

cbourjau opened the pull request:
### Description

This PR enables an easy, reproducible, and cross-platform (this PR ships linux-64 and osx-arm64 support, but Windows can be added, too) conda-forge-based developer environment using the [Pixi](https://pixi.sh/latest/) package manager. A CI workflow testing that onnx can be correctly built on MacOS arm64 and Linux x86-64 using recent and minimal dependencies is also included.

### Motivation and Context

#### Improve developer experience

Sporadically contributing to onnx has been a pain point for me and my colleagues in the past. The issues are primarily centered on the following:
- No easily reproducible developer environment that is known to work (based on lock files). This means compatible (non-Python) dependencies such as protobuf, cmake, various linters, or compiler toolchains must be manually maintained and updated. 
- Required CMake flags and build commands are easy to forget and not easy to discover. Examples in this regard are running the gtest suite or re-rendering the operators' docs for the ai.onnx.ml domain.

Conda-forge is an established ecosystem, especially in data science and academic communities. Pixi is a cross-platform, state-of-the-art package manager based on Conda-forge. It allows for creating reproducible developer environments based on a lock file and enables simple orchestration tasks based on "tasks". See `INSTALL.md` edits for further details.

#### CI testing

We have run into issues (a recent [example](https://github.com/conda-forge/onnx-feedstock/pull/131)) building onnx for conda-forge in the past. While this PR does not guarantee that such issues won't occur in the future, it would certainly help to ensure in the CI of this project that things still work on the conda-forge infrastructure. Pixi is very well suited for CI tasks. It allows for the efficient definition of environments with different dependencies which can then be used in CI. For instance, this PR ensures that onnx can be correctly built in an environment with (locked) old dependencies ("oldies" environment) and (locked) recent dependencies ("default" environment). 

While not a replacement for ManyLinux-based PyPI builds, the CI integration in this PR may serve as a starting point to further simplify CI workflows.

andife commented on pixi.toml:
```diff
@@ -0,0 +1,83 @@
+packaging = ">=24.1"
+parameterized = ">=0.9.0"
+pillow = ">=11.2.1,<12"
+pip = ">=25.0"
```
the dependency list could probably improved: twine is not needed any more? in our other pipeline we use trusted publishing

cbourjau commented:
I cleaned up the dependencies as pointed out in the review. I also added support for win-64, which is now also tested on CI. I furthermore removed the linter runs from CI so that we don't get annoying CI failures if linters are updated in the `requirements.txt` files and the ones defined here go out of date. The goal for now is to update this lockfile on demand. A possible follow-up to this PR is to set up a dependabot to update the lockfile using [this](https://pixi.sh/v0.45.0/integration/ci/updates_github_actions/#using-reusable-workflows) GitHub action.

cbourjau commented on pixi.toml:
```diff
@@ -0,0 +1,91 @@
+channels = ["conda-forge"]
+description = "Open Neural Network Exchange"
+name = "onnx"
+platforms = ["osx-arm64", "linux-64", "win-64"]
```
Just for clarity and to make sure we are on the same page and on the risk of being redundant: conda-forge (the package standard) is not related to PyPI. The former is an alternative and superset (e.g. conda-forge is language agnostic) of the latter. Pixi does offer compatibility with PyPI in the sense that it can mix the two. Generally, it is recommended to stay entirely within conda-forge. However, that was impossible here since some onnx (dev-)dependencies are only published to PyPI.

I, almost all of QuantCo, and its surrounding bubble work virtually exclusively in the conda-forge universe. The onnx package on conda-forge is currently built for linux-64, linux-aarch64, linux-ppc64le, osx-64, osx-arm64, win-64. The linux-ppc64le seems a bit esoteric to me at this point, but all the other architectures seem mainstream enough. The others seem to be already covered on PyPI. I think conda-forge will eventually add a Windows arm64 build, too. 

tl;dr: I think the set of platforms supported on PyPI is perfectly fine!

justinchuby commented on pixi.toml:
```diff
@@ -0,0 +1,91 @@
+cmd = "python onnx/defs/gen_doc.py"
+
+[tasks.gtest]
+# Windows uses a different path
```
Just curious: which part of this config specifies it’s for windows?
